skglm/solvers/cd_solver.py

```diff
@@ -134,23 +124,23 @@ def cd_solver_path(X, y, datafit, penalty, alphas=None,
         else:
             if coef_init is not None:
                 w = coef_init.copy()
-                supp_size = penalty.generalized_support(w).sum()
+                supp_size = penalty.generalized_support(w[:n_features]).sum()
                 p0 = max(supp_size, p0)
                 if supp_size:
-                    Xw = X @ w
+                    Xw = X @ w[:n_features] + fit_intercept * w[-1]
                 # TODO explain/clean this hack
                 else:
                     Xw = np.zeros_like(y)
             else:
-                w = np.zeros(n_features, dtype=X.dtype)
+                w = np.zeros(n_features + fit_intercept, dtype=X.dtype)
                 Xw = np.zeros(X.shape[0], dtype=X.dtype)
 
         sol = cd_solver(
-            X, y, datafit, penalty, w, Xw,
+            X, y, datafit, penalty, w, Xw, fit_intercept=fit_intercept,
             max_iter=max_iter, max_epochs=max_epochs, p0=p0, tol=tol,
             use_acc=use_acc, verbose=verbose, ws_strategy=ws_strategy)
 
-        coefs[:, t] = w
+        coefs[:, t] = sol[0]
```

Comment:
`w` should still work, because it is of size`n_features + fit_intercept`. 
Otherwise it means we have no way to warm start the intercept, which is not great